src/main/python/mlsearch/api_requester.py

```diff
@@ -0,0 +1,161 @@
+from mlsearch.config import Config
+from mlsearch.protocol import Protocol
+from github import Github
+from requests.auth import HTTPBasicAuth
+import json
+import requests
+# import scholarly
+
+
+class APIRequest():
+    """For handling the different Valid API requests."""
+
+    def __init__(self, source, query, init_idx, count):
+        """
+        Initialization for the class.
+
+        :param  source:     The API request destination.
+        :param  query:      The query for searching.
+        :param  init_idx:   The initial pagination index.
+        :param  count:      The number of records to be fetched.
+        """
+
+        self.params = {'query':query, 'init_idx':init_idx, 
+                            'count':count, 'source': source}
+        self.params_model = {'query':str, 'init_idx':int, 
+                                  'count':int}
+        # Load the configuration file
+        self._config = Config
+        # Validate Params
+        self._validate_params()
+        # Response data
+        self.data = {'response_code': 201, 'content': None}
+
+    @property
+    def github_acc_token(self):
+        return self._config.GITHUB_ACC_TOKEN
+
+    @github_acc_token.setter
+    def github_acc_token(self, access_token):
+        if access_token:
+            self._config.GITHUB_ACC_TOKEN = access_token
+
+
+    @property
+    def pwc_auth_info(self):
+        return (self._config.PWC_USER_NAME, self._config.PWC_PASSWORD)
+
+    @pwc_auth_info.setter
+    def pwc_auth_info(self, auth_info: "tuple(user_name, password)"):
+        assert isinstance(auth_info, tuple), \
+           f"Invalid type for auth_info. Expected tuple but got {type(auth_info)}." 
+        if len(auth_info) == 2:
+            assert isinstance(auth_info[0], str), \
+                f"Invalid type for user_name. Expected str but got {type(auth_info[0])}."
+            assert isinstance(auth_info[1], str), \
+                f"Invalid type for password. Expected str but got {type(auth_info[1])}."
+            self._config.PWC_USER_NAME = auth_info[0]
+            self._config.PWC_PASSWORD = auth_info[1]
+        else:
+            raise AttributeError(f"Expected tuple with length 2 but got {len(auth_info)}.")
+
+    def _validate_params(self):
+        """Validate user input data."""
+
+        for item, typ in self.params_model.items():
+            if item in self.params.keys():
+                if not typ == type(self.params[item]):
```

Comment:
I think we can make this into 1 condition only. How do you think?

Reply:
You are right. Thanks.